polygon/bridge/db.go

```diff
@@ -0,0 +1,148 @@
+package bridge
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"time"
+
+	"github.com/ledgerwatch/erigon-lib/kv"
+	"github.com/ledgerwatch/erigon/polygon/heimdall"
+)
+
+// GetLatestEventID the latest state sync event ID in given DB, 0 if DB is empty
+// NOTE: Polygon sync events start at index 1
+func GetLatestEventID(db kv.RoDB) (uint64, error) {
+	var eventID uint64
+	err := db.View(context.Background(), func(tx kv.Tx) error {
+		cursor, err := tx.Cursor(kv.PolygonBridge)
```

Comment:
I think you need to close the cursor in a defer, ie `defer cursor.Close()` - can you please double check the other methods in here too? I think they might be missing that as well